sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveStrategies.scala

```diff
@@ -65,15 +65,20 @@ private[hive] trait HiveStrategies {
         // hive table scan operator to be used for partition pruning.
         val partitionKeyIds = AttributeSet(relation.partitionKeys)
         val (pruningPredicates, otherPredicates) = predicates.partition { predicate =>
-          !predicate.references.isEmpty &&
+          predicate.references.nonEmpty &&
           predicate.references.subsetOf(partitionKeyIds)
         }
+        val additionalPartPredicates =
+          PhysicalOperation.partitionPrunningFromDisjunction(
+            otherPredicates.foldLeft[Expression](Literal(true))(And(_, _)), partitionKeyIds)
 
         pruneFilterProject(
           projectList,
           otherPredicates,
           identity[Seq[Expression]],
-          HiveTableScanExec(_, relation, pruningPredicates)(sparkSession)) :: Nil
+        HiveTableScanExec(_,
+            relation,
+            pruningPredicates ++ additionalPartPredicates)(sparkSession)) :: Nil
```

Comment:
Sure, we understand that the additionalPartPredicates is the partition filter. But we may not be able to assure BooleanSimplification will push all NOT operator to leaf expression, as BooleanSimplification is an "optimizer" rule, which can be skipped if exceeding max iterations during optimization.